### ISSUE TITLE
Unable to install CNproScan

### ISSUE THREAD

**Issue by ChristyPeterson** [robinjugas/CNproScan#3]
Hi,

I'm trying to install CNproScan in a conda environment and I'm running into issues when I try to install the CNproScan environment. I've installed the following conda packages in the environment:
r-essentials (includes r-foreach)
parallel
r-doparallel
r-seqinr
bioconductor-rsamtools
bioconductor-genomicranges
bioconductor-iranges

When I get into the environment and enter the R console, I'm able to install devtools no problem, but when trying to install CNproScan I get the following error:
```> devtools::install_github("robinjugas/CNproScan")
Downloading GitHub repo robinjugas/CNproScan@HEAD

Skipping 2 packages not available: GenomicRanges, Rsamtools
* checking for file /tmp/RtmpimxL0z/remotes3de830d19259/robinjugas-CNproScan-e006600/DESCRIPTION ... OK
* preparing CNproScan:
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building CNproScan_0.0.0.9000.tar.gz
Warning: invalid uid value replaced by that for user 'nobody'
Warning: invalid gid value replaced by that for user 'nobody'

* installing *source* package ‘CNproScan’ ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
Error in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]) :
  there is no package called ‘GenomeInfoDbData’
Calls: <Anonymous> ... loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
ERROR: lazy loading failed for package ‘CNproScan’
* removing ‘~/conda-envs/CNV/lib/R/library/CNproScan’
Warning message:
In i.p(...) :
  installation of package '/tmp/RtmpimxL0z/file3de8207b5cc5/CNproScan_0.0.0.9000.tar.gz' had non-zero exit status
```  

Should I try to install that package that it refers to as missing (GenomeInfoDbData)? If so, I think the readme needs to be edited to include it as a dependency.

**Comment by ChristyPeterson** [comment]
Probably should have checked the environment before posting. Attempting to install the `bioconductor-genomeinfodbdata` conda package indicates that its already installed.

**Comment by robinjugas** [comment]
Hello. Thank you for that hint. I'll look into that issue. 
Does it work or not at all?

**Comment by ChristyPeterson** [comment]
I guess my question is, if I've installed the packages mentioned in conda, do I then that to install them in the R console? Perhaps that is why CNproScan wont finish installing?

**Comment by robinjugas** [comment]
I had success with this conda definition:
channels:
  - bioconda
  - conda-forge
  - defaults
dependencies:
  - conda-forge::r-base>=4.0.0
  - conda-forge::r-devtools
  - conda-forge::r-foreach
  - conda-forge::r-doparallel
  - conda-forge::r-seqinr
  - conda-forge::r-mgcv
  - bioconda::bioconductor-rsamtools
  - bioconda::bioconductor-genomicranges
  - bioconda::bioconductor-iranges

And then R -e 'devtools::install_github("robinjugas/CNproScan")'

**Comment by robinjugas** [comment]
> I guess my question is, if I've installed the packages mentioned in conda, do I then that to install them in the R console? Perhaps that is why CNproScan wont finish installing?

You have to be sure, your R is pointing to the R inside the conda env. Then, I think you don't need to install then again.

**Comment by ChristyPeterson** [comment]
Sorry for the delay, I was able to set up the same environment as you described. Everything completed, and I confirmed all the packages were installed and R is pointing to the proper conda dir. 

When I went to install your package, I got the following error:
```
> devtools::install_github('JEFworks-Lab/veloviz')
Error: Failed to install 'unknown package' from GitHub:
  HTTP error 404.
  Not Found

  Did you spell the repo owner (`JEFworks-Lab`) and repo name (`veloviz`) correctly?
  - If spelling is correct, check that you have the required permissions to access the repo.
```
I followed this example [here](https://github.com/JEFworks-Lab/veloviz/issues/5#issuecomment-758002833) and was able to initiate the install using 'remotes' instead of devtools.
`> remotes::install_github('robinjugas/CNproScan')`

However, the error I posted above remains the same, it fails the install due to a lack of package 'GenomeInfoDbData'. I wonder if there are permission issues with my conda environments? Something more to look into but perhaps an issue on my end.
